tests/codegen/scratch_buffer.cpp

```diff
@@ -65,6 +65,36 @@ TEST(codegen, str_stack)
                                LARGE_ON_STACK_LIMIT);
 }
 
+TEST(codegen, map_value_int_scratch_buf)
+{
+  test_stack_or_scratch_buffer("kprobe:f { @x = 1; @y = @x }",
+                               NAME,
+                               SMALL_ON_STACK_LIMIT);
+}
+
+TEST(codegen, map_value_int_stack)
+{
+  test_stack_or_scratch_buffer("kprobe:f { @x = 1; @y = @x }",
+                               NAME,
+                               LARGE_ON_STACK_LIMIT);
+}
+
+TEST(codegen, map_value_tuple_scratch_buf)
```

Comment:
nit: Maybe add a comment about the fact that these need an allocation and maybe global scratch buf because these are different sized tuples.

Reply:
Fixed in latest